MetaMask/metamask-extension
# fix: Prevent requests to the network provider until after onboarding is complete

…mplete

## **Description**



## **Related issues**

Fixes: https://github.com/MetaMask/metamask-planning/issues/881

## **Manual testing steps**

1. Add `await new Promise((resolve) => setTimeout(resolve, 10000));` to the top of initialize on the background.js file
2. Build the app
3. Load extension and immediately open the network tab on the background console before the app initializes
4. No requests should happen

## **Screenshots/Recordings**



### **Before**



https://github.com/MetaMask/metamask-extension/assets/7499938/73513327-90fb-4e9f-8af2-aa41c731b647





### **After**


https://github.com/MetaMask/metamask-extension/assets/7499938/d6dee1c6-fa13-48c8-9fe2-520e3d43d0b3



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
